docs/configuration.md

```diff
@@ -428,6 +428,37 @@ insert newlines.
 > You might be interested in the [Vertical Multiline](#vertical-multiline)
 > section.
 
+### `newlines.source`
+
+This parameter controls the general approach to line breaks, and whether to take into
+account existing newlines in the source. The default value (if the parameter is not
+specified) is the _classic_, original way. Below are the alternatives.
+
+> These alternatives are EXPERIMENTAL and might change in the future without
+> regard to any `edition` settings, until fully released (and this message deleted).
+
+#### `newlines.source=keep`
+
+This approach attempts to preserve line breaks in the input whenever possible.
+
+#### `newlines.source=fold,unfold`
+
+These two approaches _completely ignore_ existing line breaks, except around comments
+and blank lines (i.e., multiple consecutive newlines).
+
+> Might require increasing runner limits
+> (`runner.optimizer.maxVisitsPerToken`, possibly even `runner.maxStateVisits`),
```

Comment:
It will be more convenient to show specific values here, so that users can simply copy them